rustls/src/msgs/handshake.rs

```diff
@@ -302,6 +307,12 @@ impl ConvertServerNameList for [ServerName] {
             .filter_map(only_dns_hostnames)
             .next()
     }
+
+    fn standards_complaint(&self) -> bool {
```

Comment:
I think you meant `_compliant`? Might be nice to have a docstring that explains the issue + maybe some links to the spec and/or issues we've seen.

I would probably prefer for this to be inlined at its single callsite.